runelite-client/src/main/java/net/runelite/client/plugins/wikilookup/WikiLookupPlugin.java

```diff
@@ -0,0 +1,175 @@
+
+	@Inject
+	private MenuManager menuManager;
+
+	@Inject
+	private WikiLoopupInput inputListener;
+
+	@Inject
+	private KeyManager keyManager;
+
+	private boolean hotKeyPressed = false;
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		keyManager.registerKeyListener(inputListener);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		keyManager.unregisterKeyListener(inputListener);
+	}
+
+	@Subscribe
+	public void onMenuObjectClicked(MenuOptionClicked click)
+	{
+		if (click.getMenuOption().equals(LOOKUP))
+		{
+			String ourLookupName = Text.removeTags(click.getMenuTarget()).replaceAll("\\(.*?\\)", "").replace(' ', '_');
+			LinkBrowser.browse("http://oldschoolrunescape.wikia.com/wiki/" + ourLookupName);
+		}
+	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!hotKeyPressed)
+		{
+			return;
+		}
+
+		if (event.getType() == MenuAction.EXAMINE_ITEM_GROUND.getId()
+			|| event.getType() == MenuAction.EXAMINE_ITEM.getId()
+			|| event.getType() == MenuAction.EXAMINE_ITEM_BANK_EQ.getId()
+			|| event.getType() == MenuAction.EXAMINE_OBJECT.getId()
+			|| event.getType() == MenuAction.EXAMINE_NPC.getId()
+			|| (event.getType() == MenuAction.WIDGET_DEFAULT.getId() && Text.removeTags(event.getOption()).toLowerCase().equals("read journal:")))
+		{
+			//Change the option to Lookup
+			changeOption(event.getType(), LOOKUP);
+		}
+		else if (event.getType() == MenuAction.WIDGET_DEFAULT.getId() && Text.removeTags(event.getOption()).toLowerCase().contains("open"))
+		{
+			changeOption(event.getType(), LOOKUP);
+			changeTarget(event.getType(), Text.removeTags(event.getOption().replaceFirst("Open ", "").replaceAll("Journal", "") + "Diary"));
+		}
+		else if ((event.getType() == MenuAction.WIDGET_TYPE_2.getId() || event.getType() == MenuAction.WIDGET_DEFAULT.getId()) && Text.removeTags(event.getOption()).toLowerCase().equals("cast"))
+		{
+			addBlankOption("cast", LOOKUP);
+		}
+	}
+
+	private void changeOption(int actionID, String newOption)
+	{
+		MenuEntry[] entries = client.getMenuEntries();
+
+		for (int i = 0; i < entries.length; i++)
+		{
+			if (entries[i].getType() == actionID)
+			{
+				entries[i].setOption(newOption);
+				client.setMenuEntries(entries);
+				return;
+			}
+		}
+	}
+
+	private void changeTarget(int actionID, String newTarget)
+	{
+		MenuEntry[] entries = client.getMenuEntries();
+
+		for (int i = 0; i < entries.length; i++)
+		{
+			if (entries[i].getType() == actionID)
+			{
+				entries[i].setTarget(newTarget);
+				client.setMenuEntries(entries);
+				return;
+			}
+		}
+	}
+
+	private void addBlankOption(String toCopy, String newOption)
+	{
+		MenuEntry[] entries = client.getMenuEntries();
+		MenuEntry[] newEntries = new MenuEntry[entries.length + 1];
+		for (int i = 0; i < entries.length; ++i)
+		{
+			if (Text.removeTags(entries[i].getOption()).toLowerCase().contains(toCopy))
+			{
+				newEntries[newEntries.length - 1] = MenuEntry.builder()
```

Comment:
Instead of manually copying the entire entry, which is prone to bugs when new fields get added to the class, you could add `@Builder(toBuilder=true)` on `MenuEntry`, which allows you to do `entries[i].toBuilder().option(newOption).build()` here.

Reply:
Specifically here we want to copy the target, option, and type. We need the option to be of the same type so that we can get the name of the thing its used on.

Reply:
So this change would allow you to build a MenuEntry from an existing entry but only change certain aspects? Awesome. I will work on it.